### PR TITLE
#70 compute complete solr properties field

### PR DESCRIPTION
**Pull request by spacemansteve**
read in 8 additional files during nonbib ingest and set properties field accordingly

### CODE REVIEW

File: adsdata/nonbib.py [hunk 1]
```diff
@@ -170,6 +175,14 @@ def get_changed_rows_table(self, table_name, schema_name, meta=None):
                      Column('id', Integer),
                      Column('authors', ARRAY(String)),
                      Column('refereed', Boolean),
+                     Column('ads_openaccess', Boolean),
```
Comment (romanchyla) [inline adsdata/nonbib.py]: I think you might agree that with every new feature this query is getting too long; and we should really use ORM while we can. But there is an issue for that already, so I'm just adding a comment here

(https://github.com/adsabs/AdsDataSqlSync/issues/59)

File: run.py [hunk 2]
```diff
@@ -155,6 +180,23 @@ def add_data_link(session, current_row):
     result = session.execute(q)
     current_row['data_links_rows'] = fetch_data_link_record(result.fetchall())
 
+def is_article(property):
+    """is the passed property value for an article or nonarticle
+
+    passed an array of strings, the current property value for the bibcode"""
+    article_types = set(['eprint', 'article', 'inproceedings', 'inbook'])
```
Comment (romanchyla) [inline run.py]: a small detail, but since you are going to call this many millions times, article_types should be initialized once

File: run.py [hunk 3]
```diff
@@ -165,10 +207,11 @@ def nonbib_to_master_pipeline(nonbib_engine, schema, batch_size=1):
     tmp = []
     i = 0
     max_rows = config['MAX_ROWS']
-    q = session.query(models.NonBibTable).options(load_only(*nonbib_to_master_fields))
+    q = session.query(models.NonBibTable).options(load_only(*nonbib_to_master_select_fields))
     for current_row in q.yield_per(100):
         current_row = nonbib_to_master_dict(current_row)
         add_data_link(session, current_row)
+        cleanup_for_master(current_row)
```
Comment (romanchyla) [inline run.py]: if you are only asking for fields that are sent to master, why is it necessary to remove them? something to do with 'coalescence'?

Comment (spacemansteve) [reply run.py]: we use many of the boolean fields to set the 'property' value that goes to master, but they are not sent to master as individual fields.  so we request them, use them, and then delete them before we construct the protobuf.

File: adsdata/nonbib.py (hunk 1 above)
Comment (spacemansteve) [reply adsdata/nonbib.py]: agreed!

**Pull request merged**